### PR TITLE
[connectivity_plus] Migrate connectivity code to connectivity_plus

### PR DESCRIPTION
**Pull request by HakkyuKim**
part of #116 

Setting the Tizen platform interface is done similarly with #120.

### CODE REVIEW

**Comment by HakkyuKim** [comment]
Trying to figure out #99.

**Comment by HakkyuKim** [comment]
> Trying to figure out #99.

I'm unable to solve #99, I have no clue why this error occurs. You have to press 'q' **instantly** when the first app screen appears. I was able to reproduce the issue only on mobile devices. The chances of users encountering this error may be slim, but I do hope to know what might be causing it. Does anyone have any ideas?

File: packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        std::make_unique<flutter::MethodChannel<flutter::EncodableValue>>(
+            registrar->messenger(), "dev.fluttercommunity.plus/connectivity",
+            &flutter::StandardMethodCodec::GetInstance());
+    m_event_channel =
+        std::make_unique<flutter::EventChannel<flutter::EncodableValue>>(
+            registrar->messenger(),
+            "dev.fluttercommunity.plus/connectivity_status",
+            &flutter::StandardMethodCodec::GetInstance());
+    method_channel->SetMethodCallHandler([this](const auto &call, auto result) {
+      this->HandleMethodCall(call, std::move(result));
```
Comment (bbrto21) [inline packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc:138]:
```suggestion
      HandleMethodCall(call, std::move(result));
```

File: packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    });
+
+    auto event_channel_handler =
+        std::make_unique<flutter::StreamHandlerFunctions<>>(
+            [this](const flutter::EncodableValue *arguments,
+                   std::unique_ptr<flutter::EventSink<>> &&events)
+                -> std::unique_ptr<flutter::StreamHandlerError<>> {
+              LOG_INFO("OnListen");
+              this->registerObsever(std::move(events));
```
Comment (bbrto21) [inline packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc:147]:
```suggestion
              registerObsever(std::move(events));
```

File: packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+              return nullptr;
+            },
+            [this](const flutter::EncodableValue *arguments)
+                -> std::unique_ptr<flutter::StreamHandlerError<>> {
+              LOG_INFO("OnCancel");
+              this->clearObserver();
```
Comment (bbrto21) [inline packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc:153]:
```suggestion
              clearObserver();
```

File: packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        result = "none";
+    }
+    return result;
+  }
+
+  void HandleMethodCall(
+      const flutter::MethodCall<flutter::EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+    ensureConnectionHandle();
+    LOG_INFO("method : %s", method_call.method_name().data());
```
Comment (bbrto21) [inline packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc:104]: Should we still care about [this](https://github.com/flutter-tizen/flutter-tizen/issues/71)?

File: packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  }
+
+  virtual ~ConnectivityPlusTizenPlugin() {
+    if (m_connection != nullptr) {
+      connection_destroy(m_connection);
+      m_connection = nullptr;
+    }
+  }
+
+  void registerObsever(
```
Comment (bbrto21) [inline packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc:43]: Could you please change method names to start with a capital letter?

File: packages/connectivity_plus/example/pubspec.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    sdk: flutter
+  flutter_test:
+    sdk: flutter
+  integration_test:
+    sdk: flutter
+  integration_test_tizen:
+    path: ../../integration_test/
+
+dependency_overrides:
+  args: ^2.0.0
```
Comment (bbrto21) [inline packages/connectivity_plus/example/pubspec.yaml:23]: I don't think this is needed anymore
```suggestion
```

File: packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include "log.h"
+
+class ConnectivityPlusTizenPlugin : public flutter::Plugin {
+ public:
+  static void RegisterWithRegistrar(flutter::PluginRegistrar *registrar) {
+    LOG_INFO("RegisterWithRegistrar");
```
Comment (bbrto21) [inline packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc:26]: Seems like an unnecessary log.

File: packages/connectivity_plus/example/pubspec.yaml (hunk 6 above)
Comment (HakkyuKim) [reply packages/connectivity_plus/example/pubspec.yaml:23]: Recent migration to Flutter 2.2.1(args 1.6.0 -> args 2.1.0) by flutter-tizen/flutter-tizen/pull/138 solved this. Thanks for noticing!

File: packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc (hunk 4 above)
Comment (HakkyuKim) [reply packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc:104]: I believe statically linking higher version libstdc++ by flutter-tizen/flutter-tizen/pull/106 solved the issue.

Comment (bbrto21) [reply packages/connectivity_plus/tizen/src/connectivity_plus_tizen_plugin.cc:104]: Oh you're right! I forgot this :smiley:

**Comment by HakkyuKim** [comment]
@bbrto21 
Thanks for the review, I applied all your suggestions.

**Review by bbrto21 — APPROVED** [review]
LGTM, latest original plugin version is 1.0.4. Are you sure it's 1.0.1?

**Comment by swift-kim** [comment]
@bbrto21 The plugins are versioned separately, so their versions have nothing to do with each other.

**Comment by HakkyuKim** [comment]
> LGTM, latest original plugin version is 1.0.4. Are you sure it's 1.0.1?

You are talking about `connectivity_plus`? 
I will also update `connectivity_plus_platform_interface`.

BTW, I should update pubspec.yaml environment right?

> Recent migration to Flutter 2.2.1(args 1.6.0 -> args 2.1.0) by [flutter-tizen/flutter-tizen/pull/138](https://github.com/flutter-tizen/flutter-tizen/pull/138) solved this. Thanks for noticing!

```diff
environment:
    sdk: ">=2.12.0 <3.0.0"
-    flutter: ">=2.0.0"
+   flutter: ">=2.2.1"
```

**Comment by bbrto21** [comment]
yes, I mean [connectivity_plus](https://pub.dev/packages/connectivity_plus) version   
```yaml
dependencies:
  connectivity_plus: ^1.0.1
```


> BTW, I should update pubspec.yaml environment right?

I think we can leave it

**Comment by HakkyuKim** [comment]
@bbrto21 
Actually, I just found out that `args` got bumped up to 2.1 in flutter sdk 2.2.0 release.

If we remove:
```yaml
dependency_overrides:
  args: ^2.0.0
```
then minimum sdk version must be 2.2.0 because older version can't build it.

We can either leave `dependency_overrides`, or remove it and update min flutter version requirement. 
I'm slightly leaning towards just leaving it as done in `connectivity_plus`, but not quite sure which is a better decision. I'll think about it.

**Comment by bbrto21** [comment]
> We can either leave `dependency_overrides`, or remove it and update min flutter version requirement.
> I'm slightly leaning towards just leaving it as done in `connectivity_plus`, but not quite sure which is a better decision. Any thoughts?

I think it's better to remove the dependency override.

File: packages/connectivity_plus/pubspec.yaml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+flutter:
+  plugin:
+    platforms:
+      tizen:
+        dartPluginClass: ConnectivityPlugin
+        pluginClass: ConnectivityPlusTizenPlugin
+        fileName: connectivity_plus_tizen_plugin.h
+
+dependencies:
+  connectivity_plus_platform_interface: ^1.0.0
```
Comment (swift-kim) [inline packages/connectivity_plus/pubspec.yaml:15]: Are you going to update this to 1.0.2?

**Review by swift-kim — APPROVED** [review]

**Comment by HakkyuKim** [comment]
OK, so I came with the following conclusion. Because `connectivity_plus_tizen_example` has the following dependencies:

- connectivity_plus_tizen_example -> connectivity_plus -> connectivity_plus_linux
- connectivity_plus_tizen_example -> integration_test -> flutter sdk

And because the error message is:

```
pub get failed (1; So, because connectivity_plus_example depends on both connectivity_plus ^1.0.1 and integration_test any from sdk, version solving failed.)
```

To run integration test with `connectivity_plus`, you need to set flutter sdk to at least 2.2.0. This doesn't have to do with `connectivity_plus_tizen`, so you can ignore my original comment.

After changing `pubspec.yaml` of example package as follows:

```diff
# connectivity_plus_tizen_example
environment:
  sdk: ">=2.12.0 <3.0.0"
-  flutter: ">=2.0.0"
+  flutter: ">=2.2.0"
```

Now the error message is more clear on how to build the app:

```
The current Flutter SDK version is 2.0.1.                               
                                                                        
Because connectivity_plus_example requires Flutter SDK version >=2.2.0, version solving failed.
Running "flutter pub get" in example...                                 
pub get failed (1; Because connectivity_plus_example requires Flutter SDK version >=2.2.0, version solving failed.)
```

File: packages/connectivity_plus/pubspec.yaml (hunk 8 above)
Comment (HakkyuKim) [reply packages/connectivity_plus/pubspec.yaml:15]: Yes, I'll update it together with the sdk version issue!

**Comment by HakkyuKim** [comment]
@bbrto21, @swift-kim 
I'll merge this if the last version change makes sense.

I had to rebase to run test, only the last commit has actually changed:
"Update dependency versions 3d4d218"

**Comment by swift-kim** [comment]
@HakkyuKim Okay, thanks!

**Comment by bbrto21** [comment]
@HakkyuKim good!

**Pull request merged**